src/sdk/namespace_fs.js

```diff
@@ -3476,6 +3513,32 @@ class NamespaceFS {
         if (latest_ver_info && latest_ver_info.version_id_str === version_id) throw error_utils.new_error_code('VERSION_MOVED', `version file moved from .versions/ ${versioned_path} to latest ${latest_version_path}, retrying`);
     }
 
+    /**
+     * _unlink_file_ignore_enoent does the following - 
+     * 1. stat the to be deleted file
+     * 2. checks that the file should be deleted based on lifecycle filter (if the flow is not lifecycle the check will be skipped)
+     * 3. calls unlinkfileat that inside checks that the path to be deleted has the same inode/fd of the file that should be deleted
+     * @param {nb.NativeFSContext} fs_context 
+     * @param {Object} params 
+     * @param {String} file_path 
+     * @param {{dir_file: nb.NativeFile, src_file: nb.NativeFile }} files 
+     */
+    async _unlink_file_ignore_enoent(fs_context, params, file_path, { dir_file, src_file }) {
```

Comment:
this function has almost the same name as native_fs_utils.unlink_ignore_enoent. should use a different name to prevent confusion